docs/src/usage.md

```diff
@@ -1,3 +1,74 @@
 # Usage
 
-TODO
+If you've not read the [Julia Base Broadcast Background](@ref) section, that may
+be helpful in understanding concepts, types, and functions discussed here.
+
+The basic usage of this package involves wrapping intermediate broadcast
+expressions with dot-calls to `lazy_broadcast`, and then working with those
+resulting objects. Let's make one now.
+
+```@example usage
+using LazyBroadcast;
+x = rand(10);
+y_lazy = lazy_broadcast.(x .+ x);
+```
+
+As you can see, `y_lazy` is a `Broadcasted` object. There are several things
+that you can do with them:
+ 
+ - `materialize` them (which executes the expression)
+ - Combine with other broadcast expressions
+ - Call functions that support `Broadcasted` objects
+
+Let's try each of these.
+
+## Materialize
+
+This is as simple as calling `Base.Broadcast.materialize`
+
+```@example usage
+y_array = Base.Broadcast.materialize(y_lazy)
+```
+Or, we can call the in-place version, `Base.Broadcast.materialize!`:
+
+```@example usage
+z_array = similar(x)
+Base.Broadcast.materialize!(z_array, y_lazy)
+z_array
+```
+
+## Combine with other broadcast expressions
+
+This is as simple as combining expressions with more dot operations:
+
+```@example usage
+z_lazy = lazy_broadcast.(x .* x);
+fused_lazy = y_lazy .+ z_lazy # Equivalent to `x .+ x .+ x .* x`
+```
+
+In this combined expression, `y_lazy .+ z_lazy` is equivalent to `x .+ x .+ x .*
+x`, which means that this entire expression is fused. If we want to further
+delay execution, we can wrap the entire result in a dot-`lazy_broadcast` call:
+
+```@example usage
+so_lazy = lazy_broadcast.(y_lazy .+ z_lazy)
+```
+Now, we have a `Broadcasted` object again.
+
+## Call functions that support 
```

Comment:
This title seems to be incomplete

Reply:
Ah, yeah, meant to write this as "Call functions that support `Broadcasted` objects"

Reply:
I'll update this.